pages/index.js

```diff
@@ -8,10 +11,122 @@ const StyledHeading = styled.h2`
 `;
 
 export default function HomePage({ goals, categoryColors }) {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [selectedGoal, setSelectedGoal] = useState(null);
+  const [newGoal, setNewGoal] = useState({
+    myGoal: "",
+    repetition: false,
+    targetPerInterval: 1,
+    interval: "day",
+    deadlineVisible: false,
+    deadline: "",
+    myNewGoals: [],
+  });
+
+  const handleOpenModal = () => {
+    setIsModalOpen(true);
+    setSelectedGoal(null);
+  };
+
+  const handleOpenModalFromListItem = (goal) => {
+    setIsModalOpen(true);
+    setSelectedGoal(goal);
+  };
+
+  const handleInputChange = (event) => {
+    const { name, value, type, checked } = event.target;
+
+    if (type === "checkbox") {
+      setNewGoal((prevState) => ({
+        ...prevState,
+        [name]: checked,
+      }));
+    } else {
+      setNewGoal((prevState) => ({
+        ...prevState,
+        [name]: value,
+        myGoal: value,
+      }));
+    }
+  };
+
+  const handleTargetPerIntervalChange = (value) => {
+    setNewGoal((prevState) => ({
+      ...prevState,
+      targetPerInterval: value,
+    }));
+  };
+
+  const handleAddGoal = (event) => {
+    event.preventDefault();
+    const myNewGoal = {
+      id: (goals.length += 1),
```

Comment:
i  think it would be better to work with something like UID. 

Later if you have many goals, delete some and add new ones, there could be two or more goals with the same ID